nifi-nar-bundles/nifi-stateless-processor-bundle/nifi-stateless-processor/src/main/java/org/apache/nifi/processors/stateless/ExecuteStateless.java

```diff
@@ -0,0 +1,1202 @@
+        } catch (final InterruptedException e) {
+            Thread.currentThread().interrupt();
+            timeoutExceeded = true;
+            optionalResult = Optional.empty();
+            trigger.cancel();
+        } finally {
+            surfaceBulletins(bulletinRepository, maxBulletinId);
+        }
+
+        if (!optionalResult.isPresent()) {
+            timeoutExceeded = true;
+        }
+
+        if (timeoutExceeded) {
+            getLogger().error("Dataflow did not complete within the allotted time of {} milliseconds for {}. Routing to timeout.", timeoutMillis, flowFile);
+            if (flowFile != null) {
+                session.transfer(flowFile, REL_TIMEOUT);
+            }
+            trigger.cancel();
+            return;
+        }
+
+        // If the datflow was not successful, log an indication of why not and transfer to failure. Then return, as the dataflow has completed and there's nothing left to do.
+        final TriggerResult triggerResult = optionalResult.get();
+        if (!triggerResult.isSuccessful()) {
+            final Optional<Throwable> failureOptional = triggerResult.getFailureCause();
+            if (failureOptional.isPresent()) {
+                final Throwable cause = failureOptional.get();
+
+                if (flowFile == null) {
+                    getLogger().error("Dataflow failed to complete successfully. Yielding.", failureOptional.get());
+                } else {
+                    getLogger().error("Dataflow failed to complete successfully for {}. Routing to failure and yielding.", flowFile, failureOptional.get());
+
+                    // Add a failure.port.name attribute
+                    if (cause instanceof FailurePortEncounteredException) {
+                        final String portName = ((FailurePortEncounteredException) cause).getPortName();
+                        session.putAttribute(flowFile, "failure.port.name", portName);
+                    }
+                }
+
+            }
+
+            if (flowFile != null) {
+                session.transfer(flowFile, REL_FAILURE);
+            }
+
+            adjustCounters(session, dataflow, " (Failed attempts)");
+            session.adjustCounter("Failed Invocations", 1, false);
+            context.yield();
+
+            return;
+        }
+
+        // Create a FlowFile in this NiFi instance for each FlowFile that was output by the Stateless dataflow.
+        // We cannot simply transfer the output FlowFiles because they belong to a different, internal session and their content may not be persisted.
+        // Therefore, we create our own FlowFile whose parent is the input FlowFile (if one exists) and then add the attributes and contents as necessary.
+        final Set<FlowFile> createdSet;
+        try {
+            createdSet = createOutputFlowFiles(optionalResult.get(), session, flowFile);
+        } catch (final IOException e) {
```

Comment:
It looks like `createOutputFlowFiles` could throw other types of exceptions, should this catch block be more generic, or is the intention for other exceptions to bubble up and avoid the transfer to failure?

Reply:
IOException is the only checked Exception. That's all that I was intending to catch here.

Reply:
Thanks for the confirmation, just wanted to confirm that any potential RuntimeExceptions should bubble up as opposed to routing to failure.